pylint/checkers/exceptions.py

```diff
@@ -357,20 +357,34 @@ def _check_catching_non_exception(self, handler, exc, part):
                                  node=handler.type,
                                  args=(exc.name, ))
 
+    def _check_try_except_raise(self, node):
+        bare_raise = False
+        handler_having_bare_raise = None
+        for handler in node.handlers:
+            if bare_raise:
+                # check that subsequent handler is not parent of handler which had bare raise.
+                if utils.is_subclass_of(utils.safe_infer(handler_having_bare_raise.type),
+                                        utils.safe_infer(handler.type)):
```

Comment:
In case of a bare `except` block, I believe that `handler.type` would be `None`. Then `utils.safe_infer` would fail here: https://github.com/PyCQA/pylint/blob/bc7d64d866065473f33d0161c3608c4784c7d106/pylint/checkers/utils.py#L816 like this: `'NoneType' object has no attribute 'infer'` which would then cause pylint to crash on such files.

I believe that when a bare except block follows an exception handler with bare raise, the `try-except-raise` warning should *not* be reported. In such cases, the bare except block catches more (all) exceptions, including the ones caught by the bare raise handler. Such a code can be used to *not* catch (i.e. immediately reraise) exceptions of only certain types, while handling all the other exception types in some other way (e.g. gracefully). I think it is a valid use case and pylint should not warn about it.

Reply:
Ok, an example was provided in the related issue #2352.